napalm/nxos/nxos.py

```diff
@@ -209,22 +210,29 @@ def _commit_merge(self) -> None:
 
     def _get_merge_diff(self) -> str:
         """
-        The merge diff is not necessarily what needs to be loaded
-        for example under NTP, even though the 'ntp commit' command might be
-        alread configured, it is mandatory to be sent
-        otherwise it won't take the new configuration - see:
-        https://github.com/napalm-automation/napalm-nxos/issues/59
-        therefore this method will return the real diff (but not necessarily what is
-        being sent by the merge_load_config()
+        Uses netutils diff_network_config to create a partial configuration with the proper hierarchy.
+        Note: the netutils utility performs the diff offline.
+
+        Returns: diff with the proper hierarchy of commands that are missing from the current config.
+        Examples:
+        Candidate configuration:
+        interface loopback0
+          ip address 10.1.4.5/32
+          ip router ospf 100 area 0.0.0.1
+
+        Base (on device) - relevant section:
+        ...
+        interface loopback0
+          ip address 10.1.4.4/32
+          ip router ospf 100 area 0.0.0.1
+        ...
+
+        Diff that respects the required command hierarchy:
+        interface loopback0
+          ip address 10.1.4.5/32
         """
-        diff = []
         running_config = self.get_config(retrieve="running")["running"]
-        running_lines = running_config.splitlines()
-        for line in self.merge_candidate.splitlines():
-            if line not in running_lines and line:
-                if line[0].strip() != "!":
-                    diff.append(line)
-        return "\n".join(diff)
+        return diff_network_config(self.merge_candidate, running_config, "cisco_nxos")
```

Comment:
I'm not entirely familiar with what's this function doing, but looking at https://netutils.readthedocs.io/en/latest/netutils/configs/index.html?highlight=diff_network_config#netutils.config.compliance.diff_network_config sounds great!

I think the linter is failing on this line (see https://github.com/napalm-automation/napalm/runs/5176237179?check_suite_focus=true) then we should be gtg! 😄

Reply:
Fixed. Should have ran autopep8. 😄

Reply:
Also `black .` please: https://github.com/napalm-automation/napalm/runs/5176322883?check_suite_focus=true 😄

Reply:
I think autopep8 solved more than it should have, breaking black. :)